nav2_docking/opennav_docking_core/include/opennav_docking_core/non_charging_dock.hpp

```diff
@@ -125,6 +125,16 @@ class NonChargingDock : public ChargingDock
     throw std::runtime_error("This dock is not a charging dock!");
   }
 
+  /**
+   * @brief Start any detection pipelines required for pose refinement.
+   */
+  virtual void startDetectionProcess() = 0;
```

Comment:
I think both of these can be initialized as empty methods, ie `{}` so that they are not required to be implemented if there is no enable/disable logic. I'd add into the doxygen that this can be implemented if the detector is able to be toggled on and off. Otherwise, do not implement.